### PR TITLE
Creating user future status data when OOO request is approved

### LINKED ISSUE #1920: Create future OOO status using /requests

### PR DESCRIPTION
**Pull request by sahsisunny**
## Issue Ticket Number
Closes #1920

## Description

In this PR, we are creating a new user future status document when OOO request is approved. This will be used to update the user status when the OOO request is active.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [x] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

OOO requested 
image

OOO request approved

image

User future staus doc created
image

</details>

**Test Coverage**

| File        | Statements Covered | Branches Covered | Functions Covered | Lines Covered | Lines Missed |
|-------------|--------------------|------------------|-------------------|---------------|--------------|
| requests.ts | 93.1               | 75               | 100               | 92.3          | 24,76,143-144|

### CODE REVIEW

**Comment by palakgupta2712** [comment]
@sahsisunny test coverage?

File: controllers/requests.ts [hunk 1]
```diff
@@ -93,6 +97,24 @@ export const updateRequestController = async (req: any, res: any) => {
...
+        const createUserFutureStatusData = {
+          requestId,
+          status: REQUEST_TYPE.OOO,
+          state: statusState.UPCOMING,
+          from,
+          endsOn: until,
+          userId: requestedBy,
+          message,
+        };
+        await createUserFutureStatus(createUserFutureStatusData);
```
Comment (palakgupta2712) [inline controllers/requests.ts:114]: variable name shouldn't have verb in it.
can we rename it as `userFutureStatusData`

**Review by palakgupta2712 — CHANGES_REQUESTED** [review]

File: controllers/requests.ts (hunk 1 above)
Comment (sahsisunny) [reply controllers/requests.ts]: Done

**Comment by sahsisunny** [comment]
> @sahsisunny test coverage?

Added

**Review by palakgupta2712 — APPROVED** [review]

**Pull request merged**